pkg/queue/cluster_queue_interface.go

```diff
@@ -50,6 +52,15 @@ type ClusterQueue interface {
 	// queue is empty.
 	Pop() *workload.Info
 
+	// RequeueWorkload pushes the workload back to ClusterQueue after scheduling failure.
+	RequeueWorkload(*workload.Info) (bool, error)
+	// AddInadmissibleIfNotPresent inserts a workload that cannot be admitted into
+	// the inadmissibleWorkloads in ClusterQueue, unless it is already in the queue.
+	AddInadmissibleIfNotPresent(*workload.Info) error
```

Comment:
We should only have the methods that the scheduler needs in the interface.

`AddInadmissibleIfNotPresent` and `RequeIfNotPresent` (previously named `PushIfNotPresent`).

The implementation of StrictFIFO should be that `AddInadmissibleIfNotPresent` just calls `RequeIfNotPresent`.

`PushOrUpdate` is what you would use to receive objects from the controllers, and that will create a `workload.Info` internally. The scheduler needs to push `workload.Info` objects as we might want to carry information for the next scheduling attempts.

Reply:
I leave the function `AddInadmissibleIfNotPresent` and `QueueInadmissibleWorkloads` only.
Call `PushIfNotPresent` in `AddInadmissibleIfNotPresent` if it's the implementation of StrictFIFO. Pls take a look.